### PR TITLE
Kusto: Add ARM template and tests

### PR DESCRIPTION
**Pull request by radennis**
Added ARM template support for the Microsoft.Kusto resource provider resources: Clusters and Databases

### CODE REVIEW

File: tools/package.json [hunk 1]
```diff
@@ -1,23 +0,0 @@
...
-    "tv4": "1.2.7",
-    "vscode-json-languageservice": "3.0.2",
-    "mocha": "^2.3.3",
-    "mocha-each": "^1.1.0"
-  },
-  "devDependencies": {
-    "istanbul": "^0.4.4",
-    "mocha": "^2.5.3"
-  }
-}
```
Comment (TianoMS) [inline tools/package.json]: why is this file being removed?

Comment (radennis) [reply tools/package.json]: @TianoMS I think one of the imports I made did it. Anyway I reverted

**Comment by radennis** [comment]
@TianoMS could you please review my PR?

File: schemas/2015-01-01/deploymentTemplate.json [hunk 2]
```diff
@@ -132,6 +132,8 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2014-04-01-preview/Microsoft.Sql.json#/resourceDefinitions/firewallrules" },
                   { "$ref": "https://schema.management.azure.com/schemas/2014-04-01-preview/Microsoft.Sql.json#/resourceDefinitions/administrators" },
                   { "$ref": "https://schema.management.azure.com/schemas/2014-04-01-preview/Microsoft.Sql.json#/resourceDefinitions/disasterrecoveryconfiguration" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-09-07-privatepreview/Microsoft.Kusto.json#/resourceDefinitions/clusters" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-09-07-privatepreview/Microsoft.Kusto.json#/resourceDefinitions/databases" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >databases [](start = 141, length = 9)

this is not defined in the schema file.

**Comment by radennis** [comment]
@TianoMS I addressed your comments, would you please take a look?

**Pull request merged**